## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 499454`
Commit: [`38e920e0d5663b6512f2b4da1ddb2488ef9a071e`](https://github.com/NixOS/nixpkgs/commit/38e920e0d5663b6512f2b4da1ddb2488ef9a071e) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/38e920e0d5663b6512f2b4da1ddb2488ef9a071e..pull/499454/head))
Merge: [`0bba136a5e9847f572816a69f76bfc504f60201f`](https://github.com/NixOS/nixpkgs/commit/0bba136a5e9847f572816a69f76bfc504f60201f)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/23270958852

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-499454-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/nx7p29hqbip6ifbr1k95xl2i9qwsgb6h-python3.13-threat9-test-bed-0.6.0 \
  /nix/store/v2fdhv56m2f7pvkmb29zkk6wbivq281z-python3.13-threat9-test-bed-0.6.0-dist \
  /nix/store/zy3y1ha85ln32pjz3nsffijrq4zx5s9d-python3.14-threat9-test-bed-0.6.0 \
  /nix/store/fyarxmrlx2fys9sszx4d6pn5lls1s0ip-python3.14-threat9-test-bed-0.6.0-dist \
  /nix/store/kk6ka3kifbyglwaxjjk3539ix20zavx7-routersploit-3.4.7 \
  /nix/store/gxh81bkcapxrlqs8xczpmhwlv8zmjdag-routersploit-3.4.7-dist
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-499454-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/qcqg74qlrngsrp6jdh6m9avjqin02xx9-python3.13-threat9-test-bed-0.6.0 \
  /nix/store/bh4xvrz8v853ffcf6jb0nkp4fn3rpj0z-python3.13-threat9-test-bed-0.6.0-dist \
  /nix/store/w2akb281yaqpglc1l346z20p39wj1az3-python3.14-threat9-test-bed-0.6.0 \
  /nix/store/2z3pdm172kjg32s00gymlrlfnjd2gsyi-python3.14-threat9-test-bed-0.6.0-dist \
  /nix/store/pz0kj8s7yvf0dc7ca1plzwc2jmpfrbf4-routersploit-3.4.7 \
  /nix/store/5n0gw741msy0imsr6qysajnyyqwr4b3j-routersploit-3.4.7-dist
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python313Packages.threat9-test-bed</li>
    <li>python313Packages.threat9-test-bed.dist</li>
    <li>python314Packages.threat9-test-bed</li>
    <li>python314Packages.threat9-test-bed.dist</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python313Packages.threat9-test-bed</li>
    <li>python313Packages.threat9-test-bed.dist</li>
    <li>python314Packages.threat9-test-bed</li>
    <li>python314Packages.threat9-test-bed.dist</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python313Packages.threat9-test-bed</li>
    <li>python313Packages.threat9-test-bed.dist</li>
    <li>python314Packages.threat9-test-bed</li>
    <li>python314Packages.threat9-test-bed.dist</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python313Packages.threat9-test-bed</summary>
<pre>  /nix/build/nix-5366-843488705/source/tests/service_mocks/test_telnet_service_mock.py:1: DeprecationWarning: telnetlib was removed in Python 3.13. Please be aware that you are currently NOT using standard &#x27;telnetlib&#x27;, but instead a separately installed &#x27;standard-telnetlib&#x27;.
    from telnetlib import Telnet

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_trash - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_empty_response - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_error - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_not_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_redirect - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_random_port - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_tcp_service_mock.py::test_tcp_service_mock_get_command_mock - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_not_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_generic - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_credentials - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_udp_service_mock.py::test_udp_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
================= 14 failed, 3 deselected, 4 warnings in 5.24s =================</pre>
</details>
<details>
<summary>python314Packages.threat9-test-bed</summary>
<pre>  /nix/build/nix-5366-843488706/source/tests/service_mocks/test_telnet_service_mock.py:1: DeprecationWarning: telnetlib was removed in Python 3.13. Please be aware that you are currently NOT using standard &#x27;telnetlib&#x27;, but instead a separately installed &#x27;standard-telnetlib&#x27;.
    from telnetlib import Telnet

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_trash - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_empty_response - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_error - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_not_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_redirect - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_random_port - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_tcp_service_mock.py::test_tcp_service_mock_get_command_mock - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_not_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_generic - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_credentials - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_udp_service_mock.py::test_udp_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
================= 14 failed, 3 deselected, 4 warnings in 4.97s =================</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python313Packages.threat9-test-bed</li>
    <li>python313Packages.threat9-test-bed.dist</li>
    <li>python314Packages.threat9-test-bed</li>
    <li>python314Packages.threat9-test-bed.dist</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.threat9-test-bed</summary>
<pre>  /nix/build/nix-13837-3202601370/source/tests/service_mocks/test_telnet_service_mock.py:1: DeprecationWarning: telnetlib was removed in Python 3.13. Please be aware that you are currently NOT using standard &#x27;telnetlib&#x27;, but instead a separately installed &#x27;standard-telnetlib&#x27;.
    from telnetlib import Telnet

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_trash - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_empty_response - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_error - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_not_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_redirect - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_random_port - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_tcp_service_mock.py::test_tcp_service_mock_get_command_mock - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_not_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_generic - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_credentials - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_udp_service_mock.py::test_udp_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
================= 14 failed, 3 deselected, 4 warnings in 1.33s =================</pre>
</details>
<details>
<summary>python314Packages.threat9-test-bed</summary>
<pre>  /nix/build/nix-13837-3202601371/source/tests/service_mocks/test_telnet_service_mock.py:1: DeprecationWarning: telnetlib was removed in Python 3.13. Please be aware that you are currently NOT using standard &#x27;telnetlib&#x27;, but instead a separately installed &#x27;standard-telnetlib&#x27;.
    from telnetlib import Telnet

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_trash - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_empty_response - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_error - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_not_found - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_redirect - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_http_scenario_service.py::test_http_scenario_service_random_port - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_tcp_service_mock.py::test_tcp_service_mock_get_command_mock - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_not_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_generic - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_authorized - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_credentials - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_telnet_service_mock.py::test_telnet_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
FAILED tests/service_mocks/test_udp_service_mock.py::test_udp_service_mock_add_banner - PermissionError: [Errno 1] Operation not permitted
================= 14 failed, 3 deselected, 4 warnings in 0.58s =================</pre>
</details>
</details>

--- a/pkgs/by-name/ro/routersploit/package.nix
+++ b/pkgs/by-name/ro/routersploit/package.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   fetchFromGitHub,
   python3,
 }:
@@ -17,16 +18,18 @@ python3.pkgs.buildPythonApplication (finalAttrs: {
 
   build-system = with python3.pkgs; [ setuptools ];
 
-  dependencies = with python3.pkgs; [
-    paramiko
-    pycryptodome
-    pysnmp
-    requests
-    setuptools
-  ];
-
-  # Tests are out-dated and support for newer pysnmp is not implemented yet
-  doCheck = false;
+  dependencies =
+    with python3.pkgs;
+    [
+      paramiko
+      pycryptodome
+      pysnmp
+      requests
+      setuptools
+    ]
+    ++ lib.optionals (pythonAtLeast "3.13") [
+      standard-telnetlib
+    ];
 
   nativeCheckInputs = with python3.pkgs; [
     pytest-xdist
@@ -45,6 +48,27 @@ python3.pkgs.buildPythonApplication (finalAttrs: {
     "tests/core/"
     "tests/test_exploit_scenarios.py"
     "tests/test_module_info.py"
+  ];
+
+  disabledTests = [
+    # AttributeError: module 'paramiko' has no attribute 'DSSKey'.
+    "test_exploit_empty_response"
+    "test_exploit_error_response"
+    "test_exploit_not_found_response"
+    "test_exploit_redirect_response"
+    "test_exploit_trash_response"
+
+    # Runs substantially slower, making this test flaky
+    "test_exploit_timeout_response"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # port conflict when running simultaneous builds
+    "test_http_scenario_service_empty_response"
+    "test_http_scenario_service_error"
+    "test_http_scenario_service_found"
+    "test_http_scenario_service_not_found"
+    "test_http_scenario_service_redirect"
+    "test_http_scenario_service_trash"
   ];
 
   meta = {

